BaseLightSwitch/BaseLightToggle.cs

```diff
@@ -1,42 +1,69 @@
-﻿using System;
-using System.Collections.Generic;
-using System.Linq;
+﻿using System.IO;
 using System.Reflection;
 using System.Text;
 using UnityEngine;
 
 namespace BaseLightSwitch
 {
-    public class BaseLightToggle : HandTarget, IHandTarget
+    public class BaseLightToggle : HandTarget, IHandTarget, IProtoEventListener
     {
+        public bool IsLightsOn;
+        private FieldInfo _isLightsOnField;
+        private SubRoot _lightSwitchSubRoot;
+
+        public BaseLightToggle()
+        {
+            // Initialize default light state (ON)
+            this.IsLightsOn = true;
+            // Get light state field
+            this._isLightsOnField = typeof(SubRoot).GetField("subLightsOn", BindingFlags.Instance | BindingFlags.NonPublic);
+            // Initialize temporary variable to null (needed by invoked function "RestoreLightState")
+            this._lightSwitchSubRoot = null;
+        }
+
+        /// <summary>Returns the path to save folder.</summary>
+        private string GetLightSwitchSavePath() => Path.Combine(Path.Combine(@".\SNAppData\SavedGames\", SaveLoadManager.main.GetCurrentSlot()), "BaseLightSwitch");
```

Comment:
On certain platform the save path may be different. There should be a property either in Unity or Subnautica code to get that path. @PrimeSonic may know more.

Reply:
Only difference is that Mac uses Unix path syntax I believe, which is basically using "/", instead of "\". Maybe there is more to do?

Reply:
Use `/` or `Path.Combine` to ensure cross platform compatibility.

Also, the save path is no longer where use to be, but instead gets handled in a temporary location and then archived somewhere else.
Use UWE's `SaveLoadManager` to get the path to this folder.
```CS
SaveLoadManager.GetTemporarySavePath()
```
Using this, you don't need to use `GetCurrentSlot()` since it gives you the complete path to where the files are.

It's what we use in [SMLHelper](https://github.com/SubnauticaModding/SMLHelper/blob/master/SMLHelper/Utility/SaveUtils.cs).

Reply:
I'm already using Path.Combine for the path.
Also the code Path.Combine(@".\SNAppData\SavedGames\", SaveLoadManager.main.GetCurrentSlot()) is working fine so I don't know what you're talking about when you say "the save path is no longer where use to be, but instead gets handled in a temporary location and then archived somewhere else".
I've tested it but maybe I'm missing something?

Reply:
https://i.imgur.com/PBGLNSK.jpg

Reply:
If it works, then it works.  
All I can say is that `SaveLoadManager.GetTemporarySavePath()` can replace this and still work  
`Path.Combine(Path.Combine(@".\SNAppData\SavedGames\", SaveLoadManager.main.GetCurrentSlot())`.

But if this already works, then whatever.

Reply:
I didn't know Path.Combine was making it syntax friendly, it's good to know. I was about to PR that "/" to "\\" replacement.

Reply:
PS: I will stick with the SaveLoadManager.GetCurrentSlot() method as data of the Base Light Switches are not temporary data like screenshots (You don't want to save temporary states of lights, only their final states when game stops).
SaveLoadManager.GetTemporarySavePath() should only be used for temporary data that you might not want to save in the end.